scripts/vim-patch.sh

```diff
@@ -312,8 +312,10 @@ submit_pr() {
   pr_message="$(printf '[RFC] vim-patch:%s\n\n%s\n' "${pr_title#,}" "${pr_body}")"
 
   if [[ $push_first -ne 0 ]]; then
-    echo "Pushing to 'origin/${checked_out_branch}'."
-    output="$(git push origin "${checked_out_branch}" 2>&1)" &&
+    local push_remote
+    push_remote="$(git config --get github.user || echo origin)"
```

Comment:
What is `github.user`?  What sets that?

Reply:
Might be my own convention :D - sorry..
I have it since forever though, and it is useful for things like that.
Do you see a better way to get the remote correctly?

Reply:
We have some support for [hub](https://github.com/github/hub) and [git-hub](https://github.com/sociomantic-tsunami/git-hub) baked in.  Maybe checking for metadata they provide before falling back to `origin` would be a good idea.

For example, git-hub sets `hub.forkremote` to the name of the user's remote and (on newer versions) `hub.upstreamremote` to the name of the project's remote.

I've never used hub, but I wouldn't be surprised if it did something similar.